cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,25 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func run(args []string, out io.Writer) error {
```

Comment:
I'd suggest run(out io.Writer, args ...string)
it's a bit more like other functions as as fmt.Fprintf and so on.

Reply:
note the function declaration changes a small amount, but the rest remains unchanged.

Reply:
I changed the order in the latest commit